### PR TITLE
Sync nothing if no segments should be loaded or released

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: longjiquan <jiquan.long@zilliz.com>
/kind improvement

### CODE REVIEW

File: internal/querynode/impl.go [hunk 1]
```diff
@@ -1224,7 +1224,7 @@ func (node *QueryNode) GetDataDistribution(ctx context.Context, req *querypb.Get
 }
 
 func (node *QueryNode) SyncDistribution(ctx context.Context, req *querypb.SyncDistributionRequest) (*commonpb.Status, error) {
-	log := log.Ctx(ctx).With(zap.Int64("collectionID", req.GetCollectionID()), zap.String("channel", req.GetChannel()))
+	log := log.Ctx(ctx).With(zap.Int64("collectionID", req.GetCollectionID()), zap.String("channel", req.GetChannel()), zap.String("actions", req.String()))
```
Comment (congqixia) [inline internal/querynode/impl.go:1227]: print request may cause log too long

File: internal/querycoordv2/observers/leader_observer.go [hunk 2]
```diff
@@ -74,7 +74,10 @@ func (o *LeaderObserver) observeCollection(ctx context.Context, collection int64
 			dists := o.dist.SegmentDistManager.GetByShardWithReplica(ch, replica)
 			needLoaded, needRemoved := o.findNeedLoadedSegments(leaderView, dists),
 				o.findNeedRemovedSegments(leaderView, dists)
-			o.sync(ctx, leaderView, append(needLoaded, needRemoved...))
+			if len(needLoaded) > 0 || len(needRemoved) > 0 {
+				o.sync(ctx, leaderView, append(needLoaded, needRemoved...))
```
Comment (yah01) [inline internal/querycoordv2/observers/leader_observer.go:78]: `sync()` do nothing if the given slice is empty, this check seems to be redundant

File: internal/querynode/impl.go (hunk 1 above)
Comment (longjiquan) [reply internal/querynode/impl.go:1227]: OK, so how about printing the length of actions?

Comment (yah01) [reply internal/querynode/impl.go:1227]: the only critical contents are segmentID, nodeID, version and the actionType

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/querycoordv2/observers/leader_observer.go (hunk 2 above)
Comment (longjiquan) [reply internal/querycoordv2/observers/leader_observer.go:78]: Sorry, I forgot to pull the latest branch. Thanks for the reminder and I'll close this pr.

**Comment by longjiquan** [comment]
`LeaderObserver.sync` already did the check internally, so close this pr.
/close

**Pull request closed without merging**